clappr/clappr-ios
# Refactoring Loader

### Goal

The objective is to change `Loaders` class responsibility in the architecture.
Instead of holding the `MediaControl` and `Options` reference, the `Loader` now is responsible for managing and holding plugins reference across the Player lifecycle.

[ ------- **Loader** ------- ]
| _containerPlugins_ 
| _corePlugins_
| _playbacks_ **changed the name from playbackPlugins to playback
| _mediaControl_ **removed
| _externalPlugins_
[---------------------------]

The `Loader` is now a Singleton and won't receive the external plugins and options in its constructor.

### How to test

- Open the player and check if all controls are available correctly without any problem
- Run all tests and make sure they pass
